rotkehlchen/chain/ethereum/modules/balancer/utils.py

```diff
@@ -0,0 +1,284 @@
+from typing import Any, Dict, List, Tuple
+
+from eth_utils import to_checksum_address
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.utils import get_ethereum_token
+from rotkehlchen.chain.ethereum.trades import AMMSwap, AMMTrade
+from rotkehlchen.constants import ZERO
+from rotkehlchen.errors import DeserializationError
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_price,
+    deserialize_timestamp,
+)
+from rotkehlchen.typing import AssetAmount, ChecksumEthAddress, Location, Price, TradeType
+
+from .typing import BalancerPool, BalancerPoolToken
+
+
+def deserialize_pool_share(
+        raw_pool_share: Dict[str, Any],
+) -> Tuple[ChecksumEthAddress, BalancerPool]:
+    """May raise DeserializationError"""
+    try:
+        user_address = raw_pool_share['userAddress']['id']
+        user_amount = deserialize_asset_amount(raw_pool_share['balance'])
+        raw_pool = raw_pool_share['poolId']
+        total_amount = deserialize_asset_amount(raw_pool['totalShares'])
+        address = raw_pool['id']
+        raw_tokens = raw_pool['tokens']
+        total_weight = deserialize_asset_amount(raw_pool['totalWeight'])
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    try:
+        user_address = to_checksum_address(user_address)
+    except ValueError as e:
+        raise DeserializationError(
+            f'Invalid ethereum address: {address} in pool userAddress.',
+        ) from e
+
+    try:
+        address = to_checksum_address(address)
+    except ValueError as e:
+        raise DeserializationError(
+            f'Invalid ethereum address: {address} in pool id.',
+        ) from e
+
+    pool_tokens = []
+    for raw_token in raw_tokens:
+        try:
+            token_address = raw_token['address']
+            token_symbol = raw_token['symbol']
+            token_name = raw_token['name']
+            token_decimals = raw_token['decimals']
+            token_total_amount = deserialize_asset_amount(raw_token['balance'])
+            token_weight = deserialize_asset_amount(raw_token['denormWeight'])
+        except KeyError as e:
+            raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+        try:
+            token_address = to_checksum_address(token_address)
+        except ValueError as e:
+            raise DeserializationError(
+                f'Invalid ethereum address: {token_address} in pool token: {token_symbol}.',  # noqa: E501
+            ) from e
+
+        token = get_ethereum_token(
+            symbol=token_symbol,
+            ethereum_address=token_address,
+            name=token_name,
+            decimals=token_decimals,
+        )
+        token_user_amount = user_amount / total_amount * token_total_amount
```

Comment:
Add a check for total_amount being zero here. If that happens you end up with a division by zero here. Same for total_weight.

Reply:
done